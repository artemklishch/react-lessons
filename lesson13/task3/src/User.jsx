import React, { Component } from 'react';

class User extends Component {
  constructor(props){
    super(props);
    this.state = {
      user: ''
    }
  }
  
  componentDidMount(){
    this.fetchUserData(this.props.match.params.userId);
  }

  // componentDidUpdate(prevProps, prevState) {
  //   if (this.state.user.login !== this.props.match.params.userId) {
  //     this.fetchUserData(this.props.match.params.userId);
  //   }
  // }

  componentDidUpdate(prevProps, prevState) {
    if (this.props.match.params.userId !== this.props.match.params.userId) {
      this.fetchUserData(this.props.match.params.userId);
    }
  }

  componentWillUnmount(){
    this.setState({
      user: ''
    });
  }

  fetchUserData = userId => {
    fetch(`https://api.github.com/users/${userId}`)
      .then(response => response.json())
      .then(userData =>
        this.setState({ user: userData })
      );
  };

  render() {
    const { avatar_url, name, location } = this.state.user;
    return (
      <div className="user">
        <img alt="User Avatar" src={avatar_url} className="user__avatar" />
        <div className="user__info">
          <span className="user__name">{name}</span>
          <span className="user__location">{location}</span>
        </div>
      </div>
    );
  }
};
export default User;