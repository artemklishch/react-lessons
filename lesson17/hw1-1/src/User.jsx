import React, { useState, useEffect } from 'react'
import { useParams } from 'react-router-dom';

const User = () => {
  const [userData, setUserData] = useState(null);
  const { userName } = useParams();
  useEffect(() => {
    fetch(`https://api.github.com/users/${userName}`)
      .then(response => {
        if (!response.ok) throw new Error('Failed to load data')
        return response.json();
      })
      .then(data => setUserData(data))
  }, [userName]);
  if (!userData) return null;
  const { avatar_url, name, location } = userData;
  return (
    <div className="user">
      <img alt="User Avatar" src={avatar_url} className="user__avatar" />
      <div className="user__info">
        <span className="user__name">{name}</span>
        <span className="user__location">{location}</span>
      </div>
    </div>
  )
}

export default User
