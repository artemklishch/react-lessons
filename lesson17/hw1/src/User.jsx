import React, { useState, useEffect } from 'react'
import { useParams } from 'react-router-dom';

const User = () => {
  const { userName } = useParams();
  const [userData, setUserData] = useState({});
  useEffect(() => {
    fetch(`https://api.github.com/users/${userName}`)
      .then(response => {
        if (!response.ok) throw new Error('Failed to load data')
        return response.json();
      })
      .then(user => {
        const { name, avatar_url, location } = user;
        setUserData({ name, avatar_url, location, });
      })
  }, [userName]);
  return (
    <div className="user">
      <img alt="User Avatar" src={userData.avatar_url} className="user__avatar" />
      <div className="user__info">
        <span className="user__name">{userData.name}</span>
        <span className="user__location">{userData.location}</span>
      </div>
    </div>
  )
}

export default User
