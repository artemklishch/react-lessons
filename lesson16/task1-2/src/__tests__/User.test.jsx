import React from 'react';
import { shallow } from 'enzyme';
import User from '../User';
import { getUsersList } from '../usersGateway';

jest.mock('../usersGateway', () => {
  return {
    getUsersList: jest.fn(() => Promise.resolve()),
  };
})

describe('User', () => {
  it('should not render if state is empty', () => {
    const wrappedComponent = shallow(<User />);
    expect(wrappedComponent.find('.user').exists()).toEqual(false);
  })

  it('should render if state has data', () => {
    const wrappedComponent = shallow(<User />);
    wrappedComponent.setState({
      user: {
        avatar_url: 'https://example.com',
        name: 'Tom',
        location: 'PlanetEarth',
      }
    });
    // expect(wrappedComponent.find('.user').exists()).toEqual(true);
    expect(wrappedComponent.find('.user')).toMatchSnapshot();
  })

  it('should render if state has data', () => {
    expect(getUsersList).toBeCalled();
  })
});