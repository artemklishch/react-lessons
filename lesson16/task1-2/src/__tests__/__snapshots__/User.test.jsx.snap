// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should render if state has data 1`] = `
<div
  className="user"
>
  <img
    alt="User Avatar"
    className="user__avatar"
    src="https://example.com"
  />
  <div
    className="user__info"
  >
    <span
      className="user__name"
    >
      Tom
    </span>
    <span
      className="user__location"
    >
      PlanetEarth
    </span>
  </div>
</div>
`;
